editoast/src/models/rolling_stock_model.rs

```diff
@@ -93,50 +93,98 @@ impl RollingStockModel {
+            pub startup_acceleration: Option<f64>,
+            pub comfort_acceleration: Option<f64>,
+            pub const_gamma: Option<f64>,
+            pub etcs_brake_params: Option<Json<Option<EtcsBrakeParams>>>,
+            pub inertia_coefficient: Option<f64>,
+            pub base_power_class: Option<Option<String>>,
+            pub mass: Option<f64>,
+            pub rolling_resistance: Option<Json<RollingResistance>>,
+            pub loading_gauge: Option<i16>,
+            pub power_restrictions: Option<Json<HashMap<String, String>>>,
+            pub energy_sources: Option<Json<Vec<EnergySource>>>,
+            pub locked: Option<bool>,
+            pub electrical_power_startup_time: Option<Option<f64>>,
+            pub raise_pantograph_time: Option<Option<f64>>,
+            pub version: Option<i64>,
+            pub supported_signaling_systems: Option<Vec<Option<String>>>,
+        }
+
+        let changeset = Inner::deserialize(deserializer)?;
+
+        if changeset.effort_curves.is_none() {
+            return Err(serde::de::Error::custom("effort_curves is required"));
         }
+
+        validate_rolling_stock::<D>(
+            changeset.effort_curves.as_ref().unwrap(),
+            changeset.electrical_power_startup_time.flatten(),
+            changeset.raise_pantograph_time.flatten(),
+        )?;
+
+        Ok(RollingStockModelChangeset {
+            railjson_version: changeset.railjson_version,
+            name: changeset.name,
+            effort_curves: changeset.effort_curves,
+            metadata: changeset.metadata,
+            length: changeset.length,
+            max_speed: changeset.max_speed,
+            startup_time: changeset.startup_time,
+            startup_acceleration: changeset.startup_acceleration,
+            comfort_acceleration: changeset.comfort_acceleration,
+            const_gamma: changeset.const_gamma,
+            etcs_brake_params: changeset.etcs_brake_params,
+            inertia_coefficient: changeset.inertia_coefficient,
+            base_power_class: changeset.base_power_class,
+            mass: changeset.mass,
+            rolling_resistance: changeset.rolling_resistance,
+            loading_gauge: changeset.loading_gauge,
+            power_restrictions: changeset.power_restrictions,
+            energy_sources: changeset.energy_sources,
+            locked: changeset.locked,
+            electrical_power_startup_time: changeset.electrical_power_startup_time,
+            raise_pantograph_time: changeset.raise_pantograph_time,
+            version: changeset.version,
+            supported_signaling_systems: changeset.supported_signaling_systems,
+        })
     }
 }
 
-pub fn validate_rolling_stock(
+pub fn validate_rolling_stock<'de, D>(
     effort_curves: &EffortCurves,
     electrical_power_startup_time: Option<f64>,
     raise_pantograph_time: Option<f64>,
-) -> std::result::Result<(), ValidationError> {
+) -> Result<(), D::Error>
```

Comment:
I saw that it's part of your TODO list. The errors are going to change, but in the meantime, we might want to stick to the less ideal, but coherent way of doing: using `crate::error::Result` which uses `InternalError`.